### PR TITLE
Update the _types module to primitives types

### LINKED ISSUE #25: Cryptography package changed hazmat primitives [BUG/Question]
**Describe the bug**
If using this package with the most recent version of cryptography, apps crash due to a change in the hazmat primitives.

**To Reproduce**
Have the following requirements

fastapi[all]
fastapi-azure-auth
uvicorn
pytest

1. pip install --no-cache-dir -r requirements.txt
2. Run the sample "getting started" app `uvicorn app:app --reload`

**Stack trace**
...
    from fastapi_azure_auth import SingleTenantAzureAuthorizationCodeBearer
  File "/home/vcap/deps/0/python/lib/python3.9/site-packages/fastapi_azure_auth/__init__.py", line 1, in <module>
    from fastapi_azure_auth.auth import (  # noqa: F401
  File "/home/vcap/deps/0/python/lib/python3.9/site-packages/fastapi_azure_auth/auth.py", line 14, in <module>
    from fastapi_azure_auth.openid_config import OpenIdConfig
  File "/home/vcap/deps/0/python/lib/python3.9/site-packages/fastapi_azure_auth/openid_config.py", line 7, in <module>
    from cryptography.hazmat._types import _PUBLIC_KEY_TYPES as KeyTypes
ModuleNotFoundError: No module named 'cryptography.hazmat._types'

**Your configuration**
Python 3.9.6
using venv

### PR DESCRIPTION
**Pull request by bmoore**
This is a change in the way the hazmat section of the cryptography module is organized. This fixes the error received when using the most up to date version of cryptography (at the date of writing).

Close #25

### CODE REVIEW

**Comment by JonasKs** [comment]
Probably need to bump cryptography requirement in `pyproject.toml`.

**Comment by JonasKs** [comment]
I can see I’ve forgot to write a contributing guide, so pipelines fail on linting. Would you do a `pre-commit install` and `pre-commit run —all-files`? That will run all linters(such as `isort` which is failing) 

Also seems like GitHub action cache is messing up. Could you reset it by changing the  `-2` to `-3` [this line](https://github.com/Intility/fastapi-azure-auth/blob/main/.github/workflows/coverage.yaml#L27)?

File: pyproject.toml [hunk 1]
```diff
@@ -28,7 +28,8 @@ classifiers = [
 python = "^3.9"
 fastapi = "^0.68.0"
 aiohttp = "^3"
-python-jose = {extras = ["cryptography"], version = "^3.3.0"}
+cryptography = "^35.0.0"
+python-jose = "^3.3.0"
```
Comment (JonasKs) [inline pyproject.toml:32]: Perfect, cryptography should be like that.Please change the python-jose back to specifying extras.
```suggestion
python-jose = {extras = ["cryptography"], version = "^3.3.0"}
```
And then please do a `poetry update` for the lockfile to update. :blush:

**Comment by bmoore** [comment]
I missed the poetry update command. Running that now. I'm not terribly familiar with this development flow. I'll reset the python-jose to reference the extra. What is the reason for keeping it as an extra? Does that define certain optional dependencies?

**Comment by JonasKs** [comment]
> I missed the poetry update command. Running that now. I'm not terribly familiar with this development flow.

No worries, you're quicker to fix things than I am to review, so this has been easy for me :rocket:   

>  I'll reset the python-jose to reference the extra. What is the reason for keeping it as an extra? Does that define certain optional dependencies?

Honestly only for clarity, since it supports multiple backends. That way there is no question which one we use.

**Comment by JonasKs** [comment]
Thank you! I'll release a new version ASAP.

**Comment by JonasKs** [comment]
Released and on PyPi: https://pypi.org/project/fastapi-azure-auth/

**Comment by bmoore** [comment]
Thank you for all the review and help getting this fix out there!

**Comment by JonasKs** [comment]
My pleasure :blush:

**Pull request merged**